dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/jdbc/JdbcCustomIconStore.java

```diff
@@ -184,4 +189,17 @@ private String buildIconQuery(
 
     return sql;
   }
+
+  private String getPaginatedQuery(
+      Pager pager, String sql, MapSqlParameterSource mapSqlParameterSource) {
+
+    sql = sql + " LIMIT :limit OFFSET :offset ";
+
+    int offset = (pager.getPage() - 1) * pager.getPageSize();
```

Comment:
in case pagination by default is not working its that your default page in the request params is 0 and not actually 1.